### PR TITLE
Add frontend resource YAML for containerized build

### PR DESCRIPTION
**Pull request by dvail**
## Description

Adds the frontend.yaml resource file for containerization of the ACS UI according to https://consoledot.pages.redhat.com/docs/dev/containerized-frontends/migration/migration.html#_step_5_create_the_frontend_resource_yaml

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.

## Test manual

TBD - Unaware of any way to test other than deploying to an ephemeral cluster

### CODE REVIEW

File: deploy/frontend.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+apiVersion: v1
+kind: Template
+metadata:
+  name: acs
+objects:
+  - apiVersion: cloud.redhat.com/v1alpha1
+    kind: Frontend
+    metadata:
+      name: acs
```
Comment (dvail) [inline deploy/frontend.yaml:9]: The `name` metadata here must match the route our assets are served under in the consoledot UI.

Inspecting the network tab shows that the route is `/apps/acs`, hence the `acs` values here.

File: deploy/frontend.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          href: "/application-services/acs/overview"
+          product: "Advanced Cluster Security"
+        - appId: "acs"
+          title: "Getting Started"
+          href: "/application-services/acs/getting-started"
+          product: "Advanced Cluster Security"
+        - appId: "acs"
+          title: "ACS Instances"
+          href: "/application-services/acs/instances"
+          product: "Advanced Cluster Security"
```
Comment (dvail) [inline deploy/frontend.yaml:33]: Nav items are converted from JSON in the chrome service backend here: https://github.com/RedHatInsights/chrome-service-backend/blob/main/static/beta/stage/navigation/application-services-navigation.json#L103

According to the docs, this is currently unused but should match the external JSON for future use.

File: deploy/frontend.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      module:
+        manifestLocation: "/apps/acs/fed-mods.json"
+        modules:
+          - id: "acs"
+            module: "./RootApp"
+            routes:
+              - pathname: /application-services/acs
```
Comment (dvail) [inline deploy/frontend.yaml:40]: Federated module config is converted from JSON in the chrome service backend here: https://github.com/RedHatInsights/chrome-service-backend/blob/main/static/beta/stage/modules/fed-modules.json#L948

Again, supposedly unused but needs to match in the future.

**Review by sachaudh — APPROVED** [review]
The YAML structure looks good to me 👍🏼

**Pull request merged**